### PR TITLE
DDPB-4628 : Professional deputy users unable to change answer on significant decisions question 

### PR DESCRIPTION
**Pull request by MiaGordon91**
## Purpose
Deputies weren't able to edit the 'significant decisions' question which resulted in the summary page not reflecting their new answer from 'no' to 'yes'.

A deputy can now edit the following question 'Have you made any significant financial decisions for [client name].' and see the answer correctly reflected on the summary page. 

Fixes DDPB-4628

## Approach
- New field added to the Report table that now saves the answer to significant decisions made, instead of relying on the form object.
- Decision trait updated to set 'reasons for no decision' to null if deputy selects yes.
- Twig template updated with new property on the summary page.
- This question doesn't appear on ndr report, but whilst checking this I ended up fixing a bug that was preventing the overview page from rendering due to a missing key value pair in the state array that tracks the status of the Debts section.
- Behat test created for decision section 

## Learning
_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about DigiDeps_

## Checklist
- [x ] I have performed a self-review of my own code
- [ ] I have updated documentation (Confluence/ADR/tech debt doc) where relevant
- [x ] I have added tests to prove my work
- [ ] The product team have approved these changes
- [ ] I have checked my work for potential security issues and refered to the [OWASP top 10](https://owasp.org/www-project-top-ten/)

### Frontend
- [ ] I have run an in-browser accessibility test (e.g. WAVE, Lighthouse)
- [ ] There are no deprecated CSS classes noted in the profiler
- [ x] Translations are used and the profiler doesn't identify any missing
- [ ] Any links or buttons added are screen reader friendly and contextually complete
- [ ] If adding GA events, I have updated or [checked](docs/runbooks/GOOGLE-ANALYTICS.md) the existing category or label values

### CODE REVIEW

File: client/src/Form/Report/DecisionExistType.php [hunk 1]
```diff
@@ -14,10 +15,14 @@ class DecisionExistType extends AbstractType
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $builder
-            ->add('hasDecisions', FormTypes\ChoiceType::class, [
-                'choices' => ['Yes' => 'yes', 'No' => 'no'],
+            ->add('significantDecisionsMade', FormTypes\ChoiceType::class, [
+                // keep in sync with API model constants
+                'choices' => [
+                    'existPage.form.choices.yes' => Report::SIGNIFICANT_DECISION_MADE,
```
Comment (Gugandeep) [inline client/src/Form/Report/DecisionExistType.php:21]: Noticed when we were pairing that the form wasn't reflecting the translation files properly. Could have been a caching issue. 

Just check it looks fine on the branch environment, and you can compare this form type with others to see if the code is right.

File: api/src/Controller/Report/DecisionController.php [hunk 2]
```diff
@@ -49,7 +46,7 @@ public function upsertDecision(Request $request)
             $this->denyAccessIfReportDoesNotBelongToUser($report);
             $decision = new EntityDir\Report\Decision();
             $decision->setReport($report);
-            $report->setReasonForNoDecisions(null);
+//            $report->setReasonForNoDecisions(null);
```
Comment (Gugandeep) [inline api/src/Controller/Report/DecisionController.php:49]: Was this meant to be removed permanently, or just commented out for testing?

File: api/src/Migrations/Version268.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version268 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return '';
```
Comment (Gugandeep) [inline api/src/Migrations/Version268.php:17]: Sorry kinda trivial, but could you add a description here?

File: client/src/Entity/Report/Report.php [hunk 4]
```diff
@@ -750,18 +746,30 @@ public function setHasContacts($value)
...
 
-        return $this->getReasonForNoDecisions() ? 'no' : 'yes';
+    public function setSignificantDecisionsMade($significantDecisionsMade)
+    {
+        $this->significantDecisionsMade = $significantDecisionsMade;
+
+        return $this;
     }
 
+//    /**
```
Comment (Gugandeep) [inline client/src/Entity/Report/Report.php:761]: Again - not sure if this code was removed temporarily for testing or needs to be removed permanently?

**Review by Gugandeep — CHANGES_REQUESTED** [review]

File: api/src/Controller/Report/DecisionController.php (hunk 2 above)
Comment (MiaGordon91) [reply api/src/Controller/Report/DecisionController.php:49]: We commented it out for testing, but I think it's pretty redundant now so I'll remove it permanently

File: api/src/Migrations/Version268.php (hunk 3 above)
Comment (MiaGordon91) [reply api/src/Migrations/Version268.php:17]: Good shout, totally missed this!

File: client/src/Form/Report/DecisionExistType.php (hunk 1 above)
Comment (MiaGordon91) [reply client/src/Form/Report/DecisionExistType.php:21]: All looks fine

File: client/src/Entity/Report/Report.php [hunk 5]
```diff
@@ -750,18 +746,30 @@ public function setHasContacts($value)
...
+//     * @return string|null
+//     */
+//    public function hasDecisions()
+//    {
+//        if (empty($this->getDecisions()) && !$this->getReasonForNoDecisions()) {
+//            return null;
+//        }
+//
+//        return $this->getReasonForNoDecisions() ? 'no' : 'yes';
+//    }
```
[the remark this answers is not in the document]
Comment (MiaGordon91) [reply client/src/Entity/Report/Report.php:771]: It was advised that I left this in for now, but I'm going to remove it as it's not used anywhere

**Review by Gugandeep — APPROVED** [review]
Looks good - Great work on this one!

**Pull request merged**